src/batch/src/task/task_execution.rs

```diff
@@ -43,6 +46,45 @@ use crate::task::BatchTaskContext;
 // Now we will only at most have 2 status for each status channel. Running -> Failed or Finished.
 const TASK_STATUS_BUFFER_SIZE: usize = 2;
 
+/// A special version for batch allocation stat, passed in another task context to report task mem
+/// usage 0 bytes at the end.
+pub async fn allocation_stat_for_batch<Fut, T, F, C>(
+    future: Fut,
+    interval: Duration,
+    mut report: F,
+    context: C,
+) -> T
+where
+    Fut: Future<Output = T>,
+    F: FnMut(usize),
+    C: BatchTaskContext,
+{
+    BYTES_ALLOCATED
+        .scope(TaskLocalBytesAllocated::new(), async move {
+            // The guard has the same lifetime as the counter so that the counter will keep positive
+            // in the whole scope. When the scope exits, the guard is released, so the counter can
+            // reach zero eventually and then `drop` itself.
+            let _guard = Box::new(114514);
+            let monitor = async move {
+                let mut interval = tokio::time::interval(interval);
+                loop {
+                    interval.tick().await;
+                    BYTES_ALLOCATED.with(|bytes| report(bytes.val()));
+                }
+            };
+            let output = tokio::select! {
+                biased;
+                _ = monitor => unreachable!(),
+                output = future => {
+                    context.record_mem_usage(0);
+                    output
```

Comment:
Report mem usage as 0 after ends immediately.

Reply:
Alternatively we can use a bool flag in task context to indicate execution ends. This will also be helpful for cleaning up tasks that have been done.